src/newcharacter.cpp

```diff
@@ -1503,8 +1531,57 @@ int set_scenario(WINDOW *w, player *u, int &points)
     ctxt.register_action("NEXT_TAB");
     ctxt.register_action("SORT");
     ctxt.register_action("HELP_KEYBINDINGS");
+    ctxt.register_action("FILTER");
+
+    bool recalc_scens = true;
+    int scens_length;
+    std::string filterstring = "";
+    std::vector<const scenario *> sorted_scens;
 
     do {
+        if (recalc_scens) {
+            sorted_scens.clear();
+            for (scenmap::const_iterator iter = scenario::begin(); iter != scenario::end(); ++iter) {
+                if (!filterstring.empty() &&
+                    !lcmatch(iter->second.gender_appropriate_name(u->male), filterstring)) {
```

Comment:
I think you don't need to test for an empty filterstring at all. `lcmatch` will return true if the filterstring is empty (essentially because `std::string::find` returns that the empty string is contained in the string). This makes the `!lcmatch(...)` false when the filterstring is empty.